app/models/log_file.rb

```diff
@@ -180,10 +180,17 @@ def destination_directory
     File.join("#{resource.zone.name}_#{resource.zone.id}", "#{resource.name}_#{resource.id}")
   end
 
+  def self.logfile_name(resource, category = "Current", date_string = nil)
+    region = MiqRegion.my_region.try(:region) || "unknown"
+    [category, "region", region, resource.zone.name, resource.zone.id, resource.name, resource.id, date_string].compact.join(" ")
```

Comment:
Why the `.compact`? Could anything be nil?

Reply:
logfile_name is being used for non-logs related materials such as automate dialogs/domains.    date_string will be nil for them, but also in the unlikely event we couldn't find timestamps in the evm.log or compressed version of the evm.log.